source/compilation/builtins/SystemTasks.cpp

```diff
@@ -608,6 +608,58 @@ class SdfAnnotateTask : public SystemTaskBase {
     }
 };
 
+class PlaTask : public SystemTaskBase {
+public:
+    PlaTask(const std::string& name) :
+        SystemTaskBase(name), requiredArgs(3) {};
+
+    const Type& checkArguments(const BindContext& context, const Args& args, SourceRange range,
+                               const Expression*) const {
+        auto& comp = context.getCompilation();
+
+        if (!checkArgCount(context, false, args, range, requiredArgs, requiredArgs))
+            return comp.getErrorType();
+
+        for (size_t i = 0; i < args.size(); i++) {
+            auto& type = *args[i]->type;
+
+            if (i == 0) {
+                if (!type.isUnpackedArray()) {
+                    return badArg(context, *args[i]);
+                }
+
+                // TODO: Check that array dimensions are in ascending order
+
+                auto& arrayType = *type.getArrayElementType();
+                if (arrayType.isSimpleBitVector()) {
+                    auto range = arrayType.getCanonicalType().as<IntegralType>().getBitVectorRange();
+                    if (range.left > range.right) {
+                        context.addDiag(diag::PlaRangeInAscendingOrder, args[i]->sourceRange) << arrayType;
+                        return context.getCompilation().getErrorType();
```

Comment:
You have the compilation in local var `comp` so you can just use that here and below.